common/unsafe/src/main/java/org/apache/spark/unsafe/types/ByteArray.java

```diff
@@ -44,7 +44,7 @@ public static long getPrefix(byte[] bytes) {
       final int minLen = Math.min(bytes.length, 8);
       long p = 0;
       for (int i = 0; i < minLen; ++i) {
-        p |= (128L + Platform.getByte(bytes, Platform.BYTE_ARRAY_OFFSET + i))
+        p |= ((long) Platform.getByte(bytes, Platform.BYTE_ARRAY_OFFSET + i) & 0xff)
```

Comment:
We have [the unit test](https://github.com/kiszk/spark/blob/08776e1931ee21e5200a977efd0aa379ae448033/core/src/test/scala/org/apache/spark/util/collection/unsafe/sort/PrefixComparatorsSuite.scala#L62-L115) for this. I added some cases.